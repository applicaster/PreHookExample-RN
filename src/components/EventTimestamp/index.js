import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  Text,
} from 'react-native';
import moment from 'moment';
import {
  isWithinAWeek,
  isWithinAYear,
} from '../../utils/time';

const styles = StyleSheet.create({
  timestamp: {
    color: '#FFFFFF',
    fontSize: 12,
    bottom: -2,
    position: 'absolute',
    left: 10,
  },
});

class EventTimestamp extends Component {
  renderTime() {
    const timestamp = this.props.timestamp;
    const eventMoment = moment.unix(timestamp);

    if (isWithinAWeek(eventMoment)) {
      return eventMoment.fromNow();
    } else if (isWithinAYear(eventMoment)) {
      return eventMoment.format('MMMM D');
    }
    
    return eventMoment.format('MMMM D, YYYY');
  }
  
  render() {
    const time = this.renderTime();
    return (<Text style={styles.timestamp}>{time}</Text>);
  }
}

EventTimestamp.propTypes = {
  timestamp: PropTypes.number,
};

export default EventTimestamp;
