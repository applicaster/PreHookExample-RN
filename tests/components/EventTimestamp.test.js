import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventTimestamp from '../../src/components/EventTimestamp';

describe('EventTimestamp', () => {
  test('component renders correctly for a timestamp within a week', () => {
    const html = renderer.create(
      <EventTimestamp timestamp={1501876350} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for a timestamp within a year', () => {
    const html = renderer.create(
      <EventTimestamp timestamp={1480881148} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly for a timestamp over a year', () => {
    const html = renderer.create(
      <EventTimestamp timestamp={1449258748} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
