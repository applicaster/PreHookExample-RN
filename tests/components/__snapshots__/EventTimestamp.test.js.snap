// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTimestamp component renders correctly for a timestamp over a year 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#FFFFFF",
      "fontSize": 12,
    }
  }
>
  December 4, 2015
</Text>
`;

exports[`EventTimestamp component renders correctly for a timestamp within a week 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#FFFFFF",
      "fontSize": 12,
    }
  }
>
  August 4
</Text>
`;

exports[`EventTimestamp component renders correctly for a timestamp within a year 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#FFFFFF",
      "fontSize": 12,
    }
  }
>
  December 4
</Text>
`;
